crosschain/atomic-swap-ether/src/main/java/tech/pegasys/samples/crosschain/atomicswapether/AtomicSwapEther.java

```diff
@@ -43,11 +44,14 @@
     // the sidechains / blockchains need to be deployed at the addresses shown below,
     // with the blockchain IDs indicated.
     private static final BigInteger SC0_SIDECHAIN_ID = BigInteger.valueOf(33);
-    private static final String SC0_URI = "http://127.0.0.1:8330/";
+    private static final String SC0_IP_PORT = "127.0.0.1:8330";
+    private static final String SC0_URI = "http://" + SC0_IP_PORT + "/";
     private static final BigInteger SC1_SIDECHAIN_ID = BigInteger.valueOf(11);
-    private static final String SC1_URI = "http://127.0.0.1:8110/";
+    private static final String SC1_IP_PORT = "127.0.0.1:8110";
+    private static final String SC1_URI = "http://" + SC1_IP_PORT + "/";
     private static final BigInteger SC2_SIDECHAIN_ID = BigInteger.valueOf(22);
-    private static final String SC2_URI = "http://127.0.0.1:8220/";
+    private static final String SC2_IP_PORT = "127.0.0.1:8220";
+    private static final String SC2_URI = "http://" + SC2_IP_PORT + "/";
```

Comment:
If we start generalizing addresses instead of hardcoding them, maybe it's better to decide on a repeatable way to assign ports and apply it everywhere. For example, currently we are using for RPC the port 8000+(sidechain*10)+(node number), just because that is the scheme I came up with as a quick thing. We have no scheme nor space for devp2p ports, which soon will be needed. So I would now propose something like 50000+(sidechain * 100) + (node * 10) + (type of port), with type of port = 0 for RPC, 1 for devP2P, etc.

Reply:
I think we should have all samples inherit from a base class that sets-up the ports  etc.